pkg/automationconfig/automation_config_secret.go

```diff
@@ -11,6 +11,22 @@ import (
 
 const ConfigKey = "cluster-config.json"
 
+// ReadFromSecret returns the AutomationConfig present in the given Secret. If the Secret is not
+// found, it is not considered an error and an empty AutomationConfig is returned.
+func ReadFromSecret(secretGetter secret.Getter, secretNsName types.NamespacedName) (AutomationConfig, error) {
+	acSecret, err := secretGetter.GetSecret(secretNsName)
+	if err != nil && !apiErrors.IsNotFound(err) {
+		return AutomationConfig{}, err
+	}
+
+	// the secret not existing is perfectly fine, this will happen during the first reconciliation.
+	if apiErrors.IsNotFound(err) {
+		return AutomationConfig{}, nil
+	}
```

Comment:
[q] what do you think, for readability, to write something like:


```go
if err != nil 
{
	if apiErrors.IsNotFound(err) {
		return AutomationConfig{}, nil
	} else {
		return AutomationConfig{}, err
	}
```

We could also add a `ignoreNotFound` to `apiErrors` method similarly to: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/interfaces.go#L131